src/concrete/ml/onnx/onnx_impl_utils.py

```diff
@@ -223,3 +226,38 @@ def onnx_avgpool_compute_norm_const(
         norm_const = numpy.prod(kernel_shape)
 
     return norm_const
+
+
+def rounded_comparison(
+    x: numpy.ndarray, y: numpy.ndarray, lsbs_to_remove: int, operation: ComparisonOperationType
+) -> Tuple[bool]:
+    """Comparison operation using `round_bit_pattern` function.
+
+    `round_bit_pattern` rounds the bit pattern of an integer.
+    It also checks for any potential overflow. If so, it readjusts the LSBs accordingly.
+
+    The parameter `lsbs_to_remove` can either be an integer specifying the number of LSBS to
+    remove, or an `AutoRounder` object that determines the required number of LSBs based on the
```

Comment:
is this `AutoRounder` part true ? looking at the typing, I believe we only expect `lsbs_to_remove` to be an `int`

Reply:
then the `lsbs_to_remove`'s typing is wrong, you should also include `AutoRounder` in addition to the current `int` type ! same for the docstring

also, I can see below that you have `assert isinstance(lsbs_to_remove, int)`, so maybe you would want to remove that ! But since you then compute `half = 1 << (lsbs_to_remove - 1)`, I'm not sure I understand how `lsbs_to_remove` could be an `AutoRounder` 🤔

Reply:
No, we can't include it in this PR.

The purpose of this comment was to explain that `auto_rounding` could calculate the LSB to remove **automatically** (and indeed the best practice is to use round_bit_pattern + auto_rounding to avoid overflow problems). I mentioned at the end that in our case, we chose to calculate the LSB **manually**.

I'm sorry this comment confused you, that wasn't the point at all.

As I also mentioned, rounding_bit_battern will be replaced by truncate_bit_pattern, we will remove the workaround that calculates the lsb manually and this function will be modified to take an auto_truncate instead of an lsb (int) as input.

Reply:
yes yes I understand, thanks for explanation ! 

my initial remark was only pointing out the way the sentence is eritten : 
> The parameter `lsbs_to_remove` can either be an integer [...] or an `AutoRounder` 

I did not understand that `lsbs_to_remove` was referencing to the parameter from `round_bit_pattern` and not `rounded_comparison` since they have the same name ! I would maybe just make it more explicit then, like : 
> The parameter `lsbs_to_remove` in `round_bit_pattern` can either be an integer [...] or an `AutoRounder` 

the rest is good for me !